avro4s-core/src/main/scala/com/sksamuel/avro4s/schemas/temporal.scala

```diff
@@ -14,5 +14,5 @@ trait TemporalSchemas:
   given LocalDateSchemaFor: SchemaFor[LocalDate] = DateSchemaFor.forType
   given LocalDateTimeSchemaFor : SchemaFor[LocalDateTime] = SchemaFor(TimestampNanosLogicalType.addToSchema(SchemaBuilder.builder.longType))
   given OffsetDateTimeSchemaFor : SchemaFor[OffsetDateTime] = SchemaFor(OffsetDateTimeLogicalType.addToSchema(SchemaBuilder.builder.stringType))
-  given LocalTimeSchemaFor : SchemaFor[LocalTime] = SchemaFor(LogicalTypes.timeMicros().addToSchema(SchemaBuilder.builder.longType))
+  given LocalTimeSchemaFor : SchemaFor[LocalTime] = SchemaFor(NanosOfTheDayLogicalType.addToSchema(SchemaBuilder.builder.longType))
```

Comment:
This will break/change everyone's existing schemas I think, so we need to make this opt in.

Reply:
My understanding was that this change wouldn't break the decoding of data encoded by earlier versions. 

I now see one issue. A user could start producing the new schema and the new binary encoding without being aware of it. This could break downstream systems/users. 
Is that what you meant? 

I can check how the update was done last time.

Reply:
@sksamuel How would you make this a user opt-in? I don't see an obvious approach.

Speaking for myself, I want this breaking change on Java 17 because the current code is already broken with it.
The change in this PR won't be an issue for me because I use schema registries everywhere and there is no way to break the decoding of data serialized with earlier versions of `avro4s`.
I also don't have the downstream issue because I can update the `avro4s` version in different systems from downstream to upstream.